### DISCUSSION TITLE
How to select an MPI configuration in the latest version?

### DISCUSSION THREAD

**Discussion by h-murai** [mpi4py/mpi4py#210] [Q&A] [answered]
Hello,
I found no description about the `--mpi=` option of the _build_ command in the document of the latest version. How can I select the kind of MPI in the latest version? Do I have to directly edit `mpi.cfg`?

**Comment by dalcinl** [comment, marked as answer]
Is this documentation what you are looking for? https://mpi4py.readthedocs.io/en/stable/install.html#using-distutils

You can also run the following in the command line:
```
python setup.py build --help
...
  --mpicc            MPI C compiler command, overridden by environment
                     variables 'MPICC' (defaults to 'mpicc' if available)
  --mpi              specify a configuration section, and an optional list of
                     configuration files (e.g. --mpi=section,file1:file2), to
                     look for MPI includes/libraries, overridden by
                     environment variable 'MPICFG' (defaults to section 'mpi'
                     in configuration file 'mpi.cfg')
...
```


Typically, when doing manual installations, you have to edit `mpi.cfg`. But you SHOULD AVOID using `mpi.cfg`, this approach should be used a last resort thing. Please read the documentation linked above about using `mpicc` instead.

PS: All this may change slightly in next release. I'm slowly moving to use environment variables named `MPI4PY_BUILD_XXX` to control different aspects of the build process. The main motivation is better supporting pip install. If using pip, you do not have so much flexibility/convenience to use the command line to pass build options.

**Reply by leofang** [reply]
> If using pip, you do not have so much flexibility/convenience to use the command line to pass build options.

+1

**Comment by h-murai** [comment]
I know the document. On the other hand, there is no section of "Using distutils" in https://mpi4py.readthedocs.io/en/latest/install.html
So, `python setup.py build --mpi=foo` is still available with the latest version. Is this correct?

**Reply by dalcinl** [reply]
If you are using the in-development version (latest tag in read the docs), then yes, I removed the distutils section because distutils is deprecated by now. As I said before, I'm slowly working on refactoring some build-related stuff to make things straightforward when using `pip`. `python setup.py build --mpi=foo` is still available, and I believe it will be kept available as long as setuptools uses distutils internally.

May I give you tip? I recently added support for scikit-build. scikit-build uses CMake, so as long as CMake can find and configure MPI, things should work out of the box.

To use it, you do:

```
export MPI4PY_BUILD_BACKEND=skbuild
python setup.py  build # or `pip install .` to install
```

**Reply by h-murai** [reply]
@dalcinl Ok. Thanks.

**Reply by dalcinl** [reply]
@h-murai You may want to take a look at the docs enhancement from #218
